instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/db/SqlStatementSanitizer.java

```diff
@@ -23,12 +22,22 @@ public final class SqlStatementSanitizer {
   private static final Cache<CacheKey, SqlStatementInfo> sqlToStatementInfoCache =
       Cache.bounded(1000);
 
-  public static SqlStatementInfo sanitize(@Nullable String statement) {
-    return sanitize(statement, SqlDialect.DEFAULT);
+  public static SqlStatementSanitizer create(boolean statementSanitizationEnabled) {
+    return new SqlStatementSanitizer(statementSanitizationEnabled);
   }
 
-  public static SqlStatementInfo sanitize(@Nullable String statement, SqlDialect dialect) {
-    if (!isStatementSanitizationEnabled() || statement == null) {
+  private final boolean statementSanitizationEnabled;
+
+  private SqlStatementSanitizer(boolean statementSanitizationEnabled) {
+    this.statementSanitizationEnabled = statementSanitizationEnabled;
+  }
+
+  public SqlStatementInfo sanitizeNew(@Nullable String statement) {
+    return sanitizeNew(statement, SqlDialect.DEFAULT);
+  }
+
+  public SqlStatementInfo sanitizeNew(@Nullable String statement, SqlDialect dialect) {
```

Comment:
guessing you meant to rename this back to `sanitize`?

Reply:
🙈